### ISSUE TITLE
Camera Recording

### ISSUE THREAD

**Issue by timkendall** [fluent-ffmpeg/node-fluent-ffmpeg#351]
Has anybody used this library with recording a live video stream from a connected camera? Ideally it would be as easy as -

``` javascript
ffmpeg('/dev/video0', { timeout: 40000 })
  .format('mov')
  .save('camera-recording.mov');
```

I've used `ffmpeg -f avfoundation -i "0" ffmpeg-facetime.mpg` to record from the built in Facetime HD camera on my mac but I can't get this to work with the library.

Is this even close?

``` javascript
var command = ffmpeg({ timeout: 432000 })
  .addOption('-f', 'avfoundation')
  .addOption('-i', '"0"')
  .output('ffmpeg-factime.mpg')
  .run();
```

**Comment by crosofg** [comment]
How to end a stream recording, If I dont want to set a timeout? Is there any api call to be made?

**Comment by njoyard** [comment]
Yes @crosofg, you can call `command.kill()`.

**Comment by crosofg** [comment]
@njoyard But that does not get into the callback on"end" .

**Comment by njoyard** [comment]
Nope, that calls 'error' instead (which is the right thing to do, as ffmpeg did not terminate normally).

**Comment by crosofg** [comment]
what is the clean way to teminate the saving process?

**Comment by crosofg** [comment]
@njoyard Can you tell me the clean to terminate the process?

**Comment by njoyard** [comment]
> Yes @crosofg, you can `call command.kill()`.

This is the *only* way to terminate the process when the source does not end and no video duration was set on the command.

**Comment by Csushma08** [comment]
can any one tell me how to capture the video for the webcam through fluent ffmpeg in windows?

**Comment by LeandroLoiacono** [comment]
Hi everybody, is there a way to capture having a realtime preview in a video tag?

thanx to anybody who will answer
